python-avd/pyavd/_eos_designs/shared_utils/misc.py

```diff
@@ -427,3 +427,7 @@ def _get_ipv4_acl_field_with_substitution(field_value: str, replacements: dict[s
     @cached_property
     def ipv4_prefix_list_catalog(self: SharedUtils) -> list:
         return get(self.hostvars, "ipv4_prefix_list_catalog", default=[])
+
+    @cached_property
+    def use_router_general_for_router_id(self: SharedUtils) -> bool:
+        return get(self.hostvars, "use_router_general_for_router_id", default=False)
```

Comment:
With this you know it is always a bool, so you can remove the `is False` and `is True` on the others and just `if self.shared_utils.use_router_general_for_router_id` and  `if not self.shared_utils.use_router_general_for_router_id`
```suggestion
        return bool(get(self.hostvars, "use_router_general_for_router_id"))
```

Reply:
Done.